--- a/src/modules/simulation/gz_bridge/GZBridge.hpp
+++ b/src/modules/simulation/gz_bridge/GZBridge.hpp
@@ -44,6 +44,7 @@
 #include <uORB/PublicationMulti.hpp>
 #include <uORB/Subscription.hpp>
 #include <uORB/SubscriptionInterval.hpp>
+#include <uORB/topics/differential_pressure.h>
 #include <uORB/topics/esc_status.h>
 #include <uORB/topics/parameter_update.h>
 #include <uORB/topics/sensor_accel.h>
@@ -57,9 +58,18 @@
 #include <ignition/transport.hh>
 #include <ignition/math.hh>
 #include <ignition/msgs/imu.pb.h>
+#include <ignition/msgs/fluid_pressure.pb.h>

 using namespace time_literals;

+static constexpr auto DEFAULT_HOME_ALT_AMSL = 488.0f; // altitude AMSL at Irchel Park, Zurich, Switzerland [m]
+
+// international standard atmosphere (troposphere model - valid up to 11km) see [1]
+static constexpr auto TEMPERATURE_MSL = 288.15f; // temperature at MSL [K] (15 [C])
+static constexpr auto PRESSURE_MSL = 101325.0f; // pressure at MSL [Pa]
+static constexpr auto LAPSE_RATE = 0.0065f; // reduction in temperature with altitude for troposphere [K/m]
+static constexpr auto AIR_DENSITY_MSL = 1.225f; // air density at MSL [kg/m^3]
+
 class GZBridge : public ModuleBase<GZBridge>, public OutputModuleInterface
 {
 public:
@@ -93,12 +103,14 @@

 	void clockCallback(const ignition::msgs::Clock &clock);
 	void imuCallback(const ignition::msgs::IMU &imu);
+	void airpressureCallback(const ignition::msgs::FluidPressure &air_pressure);
 	void poseInfoCallback(const ignition::msgs::Pose_V &pose);
 	void motorSpeedCallback(const ignition::msgs::Actuators &actuators);

 	// Subscriptions
 	uORB::SubscriptionInterval _parameter_update_sub{ORB_ID(parameter_update), 1_s};

+	uORB::Publication<differential_pressure_s>	  _differential_pressure_pub{ORB_ID(differential_pressure)};
 	uORB::Publication<esc_status_s>               _esc_status_pub{ORB_ID(esc_status)};
 	uORB::Publication<vehicle_angular_velocity_s> _angular_velocity_ground_truth_pub{ORB_ID(vehicle_angular_velocity_groundtruth)};
 	uORB::Publication<vehicle_attitude_s>         _attitude_ground_truth_pub{ORB_ID(vehicle_attitude_groundtruth)};
