--- a/src/main/target/SITL/udplink.c
+++ b/src/main/target/SITL/udplink.c
@@ -61,8 +61,8 @@ int udpRecv(udpLink_t* link, void* data, size_t size, uint32_t timeout_ms) {
         return -1;
     }

-    socklen_t len;
+    int len = sizeof(link->si);
     int ret;
-    ret = recvfrom(link->fd, data, size, 0, (struct sockaddr *)&link->recv, &len);
+    ret = recvfrom(link->fd, data, size, 0, (struct sockaddr *)&link->si, &len);
     return ret;
 }
